### DISCUSSION TITLE
Page doesn't start at the top.

### DISCUSSION THREAD

**Discussion by bora-7** [thuliteio/doks#1197] [Q&A]
When I click an item on the sidebar, the page doesn't jump to the top, but instead starts like this
<img width="1439" alt="Screenshot 2024-03-03 at 15 04 39" src="https://github.com/gethyas/doks/assets/76407294/55596f5d-eab6-4b77-8c2d-1a9ad2b92071">

It doesn't do this in the demo website. How can I fix this?

**Comment by h-enk** [comment]
Does it help if you run:

```bash
npm run clean:build && npm run dev
```

**Reply by bora-7** [reply]
Thanks for the quick response. I tried it and unfortunately the issue persists:
![Uploading Screenshot 2024-03-03 at 15.32.10.png…]()

**Reply by h-enk** [reply]
Could you share your repo?

**Reply by bora-7** [reply]
I've given you access now. It's going to be open-source eventually but right now it's in early development stage so it's closed source.

**Reply by bora-7** [reply]
Do you see any possible reasons?

**Reply by h-enk** [reply]
I'll have a look tomorrow

**Reply by h-enk** [reply]
I notice this behavior only with Firefox (not Chrome, Edge, or DuckDuckGo).

And, only for the Conclusion section (the Intro and Algorithms sections work fine). When I set in `content/guide/conclusion/_index.md`

```md
---
sidebar:
  collapsed: false
---
```

I notice the behavior for all pages (Firefox only)

Will give it a thought/I'll get back to you

**Reply by h-enk** [reply]
It must be a Firefox thing — not sure what's going on here, but I got it (miraculously) working though (for me)

What I did:

1. Renamed `content/guide` to `content/docs` (the default section so to speak)
2. Set `sectionNav = ["docs"]`
3. Run `npm run clean:build && npm run dev`

All is working fine.

Then back:

1. Renamed `content/docs` to `content/guide` (the default section so to speak)
2. Set `sectionNav = ["guide"]`
3. Run `npm run clean:build && npm run dev`

And, all is working fine?!

Not a fix, I know (to be continued). But can you confirm if you're also only experiencing this behavior in Firefox and if this also works for you?

**Reply by bora-7** [reply]
Hi, yes I am using firefox. I didn't try it on other browsers yet but will do when I get to my computer. Do you mind pushing your fix to a separate branch on my repo? I'll see if it also fixes it for me and get back to you!

**Reply by h-enk** [reply]
No problem, I pushed it to `doks-1197`

**Reply by bora-7** [reply]
I just tried it. It works on google chrome, so the issue is only on Firefox as you said. I tried your branch on Firefox and the same issue persists. I also tried doing the steps you did but it unfortunately doesn't solve the issue on my machine.

**Reply by h-enk** [reply]
I really can't replicate this anymore (strangely).

Some other things, you could try: 

- Next to `npm run clean:build`, delete `hugo_stats.json` before running a new build
- Add to `assets/scss/common/_variables-custom.scss`: `$enable-smooth-scroll: false;` (there's a known Firefox scroll behavior issue)
- Add content to the pages — including a `<h2>` header (so that the ToC gets generated)
- Start with a fresh Doks install and  copy your `content` over

**Reply by bora-7** [reply]
The issue still persists on Firefox, even with trying all of those fixes. I wonder if there's anywhere I can add a Javascript script that scrolls to the top of the page on page load? It's a hacky fix but better than nothing. 

As mentioned earlier, the issue doesn't exist on other browsers, just Firefox. Thanks for your help, I'll also help investigate the issue myself by looking at the source in a few weeks when work gets easier.

Unrelated, but where can I find more on changing the colours of the theme? I couldn't find that on [this](https://getdoks.org/docs/guides/styles/) link.

**Reply by jollydodo** [reply]
I tested our website only on chrome, duckduckgo and safari which all worked perfectly fine. Then I read this topic and got curious as well with Firefox and I indeed have the same problem. Don't have time tonight to troubleshoot it further, but the behaviour is very inconsistent. Sometimes a page loads fine and sometimes it starts somewhere down on the page. Clicking around doesn't give any conclusive results yet (Thought it might be related to the content, but your pages seem empty?). Currently only 7% of our users use Firefox, but still something to consider. 

Small update: clicking the same page multiple times in a row causes it sometimes to result in the mentioned error. I then checked my console and Firefox indeed has some issues (which are not present in other browsers). Maybe this might help debug since this page triggers the error after clicking on the sidebar several times:
https://www.fablabarnhem.nl/wiki/documentatie/machines/3d-printers/itomec-i500/

This is the console error that shows up: 
"Layout was forced before the page was fully loaded. If stylesheets are not yet loaded this may cause a flash of unstyled content.". So I guess there is a problem with the loading of the stylesheets within Firefox. However, this error points to the flexsearch javascript in my case. I don't have a Netlify dev environment running right now, but could you perhaps turn off flexsearch and test again? (perhaps a dumb idea, but the best I can do right now. H-enk is the expert and probably has a better idea of what is going on ;) ).

(Unrelated but Firefox also triggers some other console errors related to the csp of utils.js and an unused preload of this font: “fonts/vendor/jost/jost-v4-latin-500.woff2” )

**Reply by bora-7** [reply]
Thanks for your response. I actually never thought of checking the console logs! But you are right, there are warnings logged there and the top one is about the layout. I've pasted the screenshot of the warnings I get below. You are also right that the top warning is the one causing this particular issue, since when the issue doesn't happen I don't see that log anymore (i still see the others).

<img width="793" alt="Screenshot 2024-03-05 at 21 34 40" src="https://github.com/gethyas/doks/assets/76407294/2d9101e9-403b-4766-b79a-0d48ad73f42b">

**Comment by bora-7** [comment]
Has there been any update on this?

**Comment by h-enk** [comment]
When you disable FlexSearch (`flexSearch = false`) it all works fine — also in Firefox.

This is the failing code: https://github.com/gethyas/doks-core/blob/main/assets/js/flexsearch.js#L121

Now, it only needs a fix — to be addressed!

**Reply by davidkopp** [reply]
Thanks. Commenting out the line solved the issue. Even without the line the search box is focused. Tested with Firefox and Edge.
